packages/pigeon/lib/objc_generator.dart

```diff
@@ -172,6 +180,20 @@ void generateObjcHeader(ObjcOptions options, Root root, StringSink sink) {
   indent.writeln('');
 
   indent.writeln('NS_ASSUME_NONNULL_BEGIN');
+
+  for (final Enum anEnum in root.enums) {
+    indent.writeln('');
+    final String enumName = _className(options.prefix, anEnum.name);
+    indent.write('typedef NS_ENUM(NSUInteger, $enumName) ');
+    indent.scoped('{', '};', () {
+      int index = 0;
+      for (final String member in anEnum.members) {
+        // Capitalized first letter to ensure Swift compatibility
+        indent.writeln('$enumName${member[0].toUpperCase()}${member.substring(1)} = $index,');
+        index++;
+      }
+    });
+  }
```

Comment:
nit: the way in which we iterate over the enums is shared across all generators.  It's pretty simple but technically someone could tweak the code here and enums would be broken.  It would be better to pull the loop into generator_tools to make sure it's shared across generators.

Reply:
I would actually prefer to have this separate. They are similar but the requirements between each are subtle, such as the trailing semicolon for last declared enum in Java or the capitalization in obj-c. The unified loop would need some generic enough way to define what to write and I don't think the added complexity there is worth the reuse of this loop.